scripts/tasks/util.py

```diff
@@ -198,3 +198,32 @@ def get_environment_from_batch_command(env_cmd, initial=None):
 
 def get_dbt_version():
     return open(get_git_root() / "toolchain" / "REQUIRED_VERSION").readline().rstrip()
+
+
+def ensure_midi_port(type, midi, port):
+    if port is None:
+        for i, p in enumerate(midi.get_ports()):
+            if str(p).startswith("Deluge"):
+                port = i
+                break
```

Comment:
the intention is for port 3 to be the dedicated sysex virtual cable, and I think this logic selects port 1 instead? maybe not critical to fix now but it would be good to do while we're here.

Reply:
I've no idea about the dedicated virtual port, but yes, this selects the first or "main" port labelled Deluge. This one works, though.

Reply:
There's cases that aren't handled on the deluge side if there's a second midi stream going at the same time. Using port 3 just mitigates that